.github/workflows/build-test-release.yml

```diff
@@ -0,0 +1,272 @@
+          export SPLUNK_BUILD_URL=https://download.splunk.com/products/${SPLUNK_PRODUCT}/releases/${SPLUNK_VERSION}/linux/${SPLUNK_LINUX_FILENAME}
+          echo $SPLUNK_BUILD_URL
+          export SPLUNK_HOME=/opt/splunk
+          wget -qO /tmp/splunk.tgz ${SPLUNK_BUILD_URL}
+          sudo tar -C /opt -zxf /tmp/splunk.tgz
+          sudo cp -r examples/data/solnlib_demo $SPLUNK_HOME/etc/apps
+          sudo cp -r solnlib $SPLUNK_HOME/etc/apps/solnlib_demo/bin/
+          sudo mkdir -p $SPLUNK_HOME/etc/apps/Splunk_TA_test/default/
+          sudo chown -R $USER:$USER /opt/splunk
+          ls $SPLUNK_HOME/etc/apps/solnlib_demo/bin/
+          echo -e "[user_info]\nUSERNAME=Admin\nPASSWORD=Chang3d"'!' | sudo tee -a /opt/splunk/etc/system/local/user-seed.conf
+          echo 'OPTIMISTIC_ABOUT_FILE_LOCKING=1' | sudo tee -a /opt/splunk/etc/splunk-launch.conf
+          sudo /opt/splunk/bin/splunk start --accept-license
+      - name: Run tests
+        run: |
+          curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python3 -
+          source $HOME/.poetry/env
+          poetry install
+          SPLUNK_HOME=/opt/splunk/ poetry run pytest --junitxml=test-results/results.xml -v examples
+      - uses: actions/upload-artifact@v2.2.4
+        with:
+          name: test-splunk test-results
+          path: test-results
+
+  publish:
+    if: github.event_name != 'pull_request'
+    needs:
+      - compliance-dependencies
+      - compliance-copyrights
+      - pre-commit
+      - review_secrets
+      - semgrep
+      - snyk
+      - build
+      - run-unit-tests
+      - test-splunk
+    runs-on: ubuntu-latest
+    env:
+      NEEDS: ${{ toJson(needs) }}
+    steps:
+      - name: check if tests have passed or skipped
+        id: check
+        shell: bash
+        run: |
+          RUN_PUBLISH=`echo $NEEDS | jq ".[] |  select(  ( .result != \"skipped\" ) and .result != \"success\" ) | length == 0"`
+          if [[ $RUN_PUBLISH != *'false'* ]]
+          then
+              echo "::set-output name=run-publish::true"
+          else
+              echo "::set-output name=run-publish::false"
+          fi
+      - name: exit without publish
+        if: ${{ steps.check.outputs.run-publish == 'false' }}
+        run: |
+          echo " some test job failed. "
+          exit 1
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+          persist-credentials: false
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v2.5.4
+        with:
+          semantic_version: 17
+          extra_plugins: |
+            @semantic-release/exec
+            @semantic-release/git
+            @google/semantic-release-replace-plugin
+        env:
+          GITHUB_TOKEN: ${{ secrets.SEMREL_TOKEN }}
+          PYPI_USERNAME: ${{ secrets.PYPI_USERNAME }}
```

Comment:
@arys-splunk I used semantic-release implementation from `addonfactory-ucc-generator` do you have access to `PYPI` creds? If yes can you also set them for this repo?

Reply:
nope, I do not. Ping Ryan please.